libs/ddhub-client-gateway-clients/src/lib/service/clients.service.ts

```diff
@@ -5,21 +5,64 @@ import {
 } from '@dsb-client-gateway/dsb-client-gateway-storage';
 import {
   ConfigDto,
+  DdhubClientsService,
   DdhubConfigService,
 } from '@dsb-client-gateway/ddhub-client-gateway-message-broker';
 import { Span } from 'nestjs-otel';
 import { MaximumNumberOfClientsReachedException } from '../exceptions/maximum-number-of-clients-reached.exception';
 import { LessThan } from 'typeorm';
+import { IamService } from '@dsb-client-gateway/dsb-client-gateway-iam-client';
 
 @Injectable()
 export class ClientsService {
   protected readonly logger = new Logger(ClientsService.name);
 
   constructor(
     protected readonly wrapper: ClientWrapperRepository,
-    protected readonly ddhubConfigService: DdhubConfigService
+    protected readonly ddhubConfigService: DdhubConfigService,
+    protected readonly ddhubClientsService: DdhubClientsService,
+    protected readonly iamService: IamService
   ) {}
 
+  @Span('clients_sync')
+  public async syncMissingClientsIds(): Promise<void> {
+    const did: string | null = this.iamService.getDIDAddress();
+
+    if (!did) {
+      this.logger.error(
+        `failing to sync clients due to not initialized iam service`
+      );
+
+      return;
+    }
+
+    const clients: string[] = await this.ddhubClientsService.getClients();
```

Comment:
Can this be a large number of clients?
If so, going to database each time to check a client exists is not very efficient, we should fetch all from the DB and then check if they are in the returned set.

Reply:
it shouldn't be much - at the end of the day it's set of strings